internal/environmentscleanup/service.go

```diff
@@ -189,39 +192,30 @@ func (s *Service) getStaleRuntimesByShoots(labelSelector string) ([]runtime, []u
 		}
 
 		log.Infof("Shoot %q is older than %f hours with age: %f hours", shoot.GetName(), s.MaxShootAge.Hours(), shootAge.Hours())
-		staleRuntime, err := s.shootToRuntime(shoot)
-		if err != nil {
-			s.logger.Infof("found a shoot without kcp labels: %v", shoot.GetName())
-			shoots = append(shoots, shoot)
+
+		instances, _, _, _ := s.instanceStorage.List(dbmodel.InstanceFilter{Shoots: []string{shoot.GetName()}})
```

Comment:
There is no index based on shoot in `instances` table so we will have len(shootList.Items) full scans in the table.
Would it be the issue?

Reply:
There are around 100 instances on dev, so I think it won't be an issue. We use this function to fetch runtimes in `kcp rt` and it's quite fast.

Reply:
This job runs only on DEV. I forgot about it.